akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -84,7 +83,11 @@ private[akka] final class InterceptorImpl[O, I](val interceptor: BehaviorInterce
 
   private def deduplicate(interceptedResult: Behavior[I], ctx: TypedActorContext[O]): Behavior[O] = {
     val started = Behavior.start(interceptedResult, ctx.asInstanceOf[TypedActorContext[I]])
-    if (started == UnhandledBehavior || started == SameBehavior || !Behavior.isAlive(started)) {
+    if (Behavior.isStopped(started)) {
+      // keep the interceptor around stopped behaviors so that it can intercept
+      // the post-stop signal
+      new InterceptorImpl(interceptor, started)
```

Comment:
Make sure we keep the interceptor when the behavior is stopped (but not for `Failed`)

Reply:
Maybe we just shouldn't do this part - this is what causes the double-Signal in persistence.